crescent/plugin.py

```diff
@@ -63,6 +65,37 @@ def load(self, path: str, refresh: bool = False) -> Plugin:
 
         return plugin
 
+    def load_folder(self, path: str, refresh: bool = False) -> list[Plugin]:
+        """Loads plugins from a folder.
+
+        ```python
+        import crescent
+
+        bot = crescent.Bot(token=...)
+
+        bot.plugins.load("project.plugin_folder")
+        ```
+
+        If a file is attempted to be loaded that does not have a plugin variable,
+        a `ValueError` will be raised. Files who's names start with an underscore
+        will not be loaded.
+
+        Args:
+            path: The path to the folder that contains the plugins.
+            refresh: Whether or not to reload the plugin and the plugin's module.
+        Returns:
+            A list of plugins that were loaded.
+        """
+
+        path = path.replace(".", os.sep)
+        loaded_plugins: list[Plugin] = []
+
+        for name in iglob(os.path.join("./", path, "**", r"[!_]*.py"), recursive=True):
+            mod_name = ".".join(name[2:-3].split(os.sep))
+            loaded_plugins.append(self.load(mod_name))
```

Comment:
This will error out if any of the files don't have a plugin (e.g. `__init__.py`). I think special files should be skipped, and then there should be a kwarg (`strict` maybe) that determines whether to just skip files without plugins.

Reply:
It doesn't error because the glob function skips those